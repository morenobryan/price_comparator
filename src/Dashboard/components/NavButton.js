import React from 'react';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';

export default ({ navigation, backgroundColor, color, iconColor, icon, text, pageName }) => {
  const styles = StyleSheet.create({
    navButton: {
      backgroundColor,
      padding: 30,
      flexDirection: 'row',
      // marginTop: -1,
      // borderBottomWidth: 1,
      // borderTopWidth: 1,
      // borderColor: '#3F51B5',
      elevation: 3,
    },
    buttonText: {
      color,
      alignSelf: 'center',
      fontFamily: 'proximaNovaAltRegular',
      fontSize: 24,
    },
    icon: {
      color: iconColor,
      alignSelf: 'center',
      fontFamily: 'flaticon',
      fontSize: 50,
      marginRight: 30,
    },
  });

  return (
    <TouchableOpacity style={styles.navButton} onPress={() => navigation.navigate(pageName)}>
      <Text style={styles.icon}>{icon}</Text>
      <Text style={styles.buttonText}>{text}</Text>
    </TouchableOpacity>
  );
};
