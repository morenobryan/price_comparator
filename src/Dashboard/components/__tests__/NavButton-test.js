import 'react-native';
import React from 'react';
import NavButton from '../NavButton';

import renderer from 'react-test-renderer';

const props = {
  navigation: {
    navigate: () => {},
  },
  backgroundColor: '#fff',
  color: '#000',
  iconColor: '#666',
  icon: 'icon',
  text: 'text',
  pageName: 'page1',
};

test('renders correctly', () => {
  const tree = renderer.create(<NavButton {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
