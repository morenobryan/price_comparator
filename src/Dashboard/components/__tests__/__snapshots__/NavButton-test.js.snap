// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "elevation": 3,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 30,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#666",
        "fontFamily": "flaticon",
        "fontSize": 50,
        "marginRight": 30,
      }
    }
  >
    icon
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#000",
        "fontFamily": "proximaNovaAltRegular",
        "fontSize": 24,
      }
    }
  >
    text
  </Text>
</View>
`;
